bindings/nodejs/lib/gpio.js

```diff
@@ -0,0 +1,116 @@
+/*
+ * This file is part of the Soletta Project
+ *
+ * Copyright (C) 2015 Intel Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+ var soletta = require( 'bindings' )( 'soletta' ),
+    _ = require( 'lodash' );
+
+exports.open = function( init ) {
+    return new Promise( function( fulfill, reject ) {
+        var pin = init.pin;
+        var dir = 0;
+        var drive_mode = 0;
+        var config = null;
+        var gpiopin;
+        var callback_data = [];
+        var edge = "any";
+
+        if ( init.pullup )
+            drive_mode = soletta.sol_gpio_drive.SOL_GPIO_DRIVE_PULL_UP;
+        else if ( init.pullup === false )
+            drive_mode = soletta.sol_gpio_drive.SOL_GPIO_DRIVE_PULL_DOWN;
+        else
+            drive_mode = soletta.sol_gpio_drive.SOL_GPIO_DRIVE_NONE;
+
+        if ( init.edge )
+            edge = init.edge;
+
+        if ( init.direction == "in" ) {
+            config = {
+                dir: soletta.sol_gpio_direction.SOL_GPIO_DIR_IN,
+                active_low: init.activeLow,
+                poll_timeout: init.poll,
+                drive_mode: drive_mode,
+                trigger_mode: edge,
```

Comment:
@sragavan @nagineni @barbieri Why not
```
trigger_mode: soletta.sol_gpio_edge_from_str( edge ),
```
... and then
```
VALIDATE_VALUE_TYPE(trigger_mode, IsUint32, "GPIO in trigger_mode", false);
config->in.trigger_mode = (sol_gpio_edge)(trigger_mode->Uint32Value());
```
[later on](https://github.com/solettaproject/soletta/pull/1817/files#diff-21ce1098458857da8b8497c2679c5f36R52)?

This would have three benefits:

0. You would have a binding for ```sol_gpio_edge_from_str()``` which is a new GPIO API that needs node.js bindings anyway.
0. You'd avoid calling one API from the bindings for another API (calling ```sol_gpio_edge_from_str()``` which is a public API from ```c_sol_gpio_config()``` which is called from ```bind_sol_gpio_open()```.
0. The problem of how to handle invalid string values such as NULL or "Chunky Monkey", would be confined to ```sol_gpio_edge_from_str()``` or, better yet and hopefully, ```sol_str_table_lookup_fallback()```.